### PR TITLE
Adds a preference for queen auto shove

### PR DESCRIPTION
**Pull request by Red-byte3D**
# About the pull request

Makes it so you can toggle the queens automatic shove, this shoulda been how it was from the start

# Explain why it's good for the game

its annoying as fuck to be a warden or stay close to the queen when you constantly get beaten up like youre in an abusive household THIS SHOULD BE ADDED TO THE KING TOO!! BUT ITS ON TM!!!

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: New preference toggle that allows Queen's to pick if they want to automatically shove people they step on or not
/:cl:

### CODE REVIEW

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by ihatethisengine** [comment]
Wouldn't it be better if autoshove just worked on disarm intent, so you could switch to it on the fly? Or at least make it a toggleable ability. I doubt anyone is going to go in the preferences every time there is a friendly pocket warden around.

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 1]
```diff
@@ -445,7 +445,7 @@
 	for(var/mob/living/carbon/xenomorph/xeno in new_loc.contents)
 		if(xeno.pass_flags.flags_pass & (PASS_MOB_THRU_XENO|PASS_MOB_THRU) && !(xeno.flags_pass_temp & PASS_MOB_THRU))
 			continue
-		if(xeno.hivenumber == hivenumber)
+		if(xeno.hivenumber == hivenumber && (queen.client?.prefs?.toggle_prefs & TOGGLE_AUTO_SHOVE))
```
Comment (ihatethisengine) [inline code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:448]:
```suggestion
		if(xeno.hivenumber == hivenumber && queen.a_intent == INTENT_DISARM)
```

Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:448]: Not a review, just an alternative suggestion.

**Comment by Red-byte3D** [comment]
> Wouldn't it be better if autoshove just worked on disarm intent, so you could switch to it on the fly? Or at least make it a toggleable ability. I doubt anyone is going to go in the preferences every time there is a friendly pocket warden around.

yes im switching to this

**Review by Drulikar — APPROVED** [review]

**Comment by Yellows91** [comment]
Does this mean when you have AUTO TACKLE set as OFF that if you swap to DISARM INTENT then you will be able to turn it on for a bit till you swap back over to any other intent?

**Comment by Drulikar** [comment]
> Does this mean when you have AUTO TACKLE set as OFF that if you swap to DISARM INTENT then you will be able to turn it on for a bit till you swap back over to any other intent?

Intent does not matter.

**Pull request merged**